lib/e2ee/qolmerrors.cpp

```diff
@@ -0,0 +1,23 @@
+// SPDX-FileCopyrightText: 2021 Carl Schwan <carlschwan@kde.org>
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+
+#include "qolmerrors.h"
+#include <cstring>
+
+Quotient::QOlmError Quotient::fromString(const char* error_raw) {
+    if (!strncmp(error_raw, "BAD_ACCOUNT_KEY", 15)) {
+        return QOlmError::BadAccountKey;
+    } else if (!strncmp(error_raw, "BAD_MESSAGE_KEY_ID", 18)) {
+        return QOlmError::BadMessageKeyId;
+    } else if (!strncmp(error_raw, "INVALID_BASE64", 14)) {
+        return QOlmError::InvalidBase64;
+    } else if (!strncmp(error_raw, "NOT_ENOUGH_RANDOM", 17)) {
+        return QOlmError::NotEnoughRandom;
+    } else if (!strncmp(error_raw, "OUTPUT_BUFFER_TOO_SMALL", 23)) {
```

Comment:
`QLatin1String` is a non-owning pointer-and-size to a C string, which is exactly what's needed here to make things less tedious.
```suggestion
Quotient::QOlmError Quotient::fromString(const char* error_raw) {
    const QLatin1String error { error_raw };
    if (error == "BAD_ACCOUNT_KEY"_ls) {
        return QOlmError::BadAccountKey;
    } else if (error == "BAD_MESSAGE_KEY_ID"_ls) {
        return QOlmError::BadMessageKeyId;
    } else if (error == "INVALID_BASE64"_ls) {
        return QOlmError::InvalidBase64;
    } else if (error == "NOT_ENOUGH_RANDOM"_ls) {
        return QOlmError::NotEnoughRandom;
    } else if (error == "OUTPUT_BUFFER_TOO_SMALL"_ls) {
```